integration_tests/src/main/python/join_test.py

```diff
@@ -101,31 +101,40 @@ def create_ridealong_df(spark, key_data_gen, data_gen, left_length, right_length
 @ignore_order(local=True)
 @pytest.mark.parametrize('join_type', ['Left', 'Inner', 'LeftSemi', 'LeftAnti'], ids=idfn)
 @pytest.mark.parametrize('batch_size', ['100', '1g'], ids=idfn)
-def test_right_broadcast_nested_loop_join_without_condition_empty(join_type, batch_size):
+@pytest.mark.parametrize('enable_aqe', [False, True], ids=idfn)
+def test_right_broadcast_nested_loop_join_without_condition_empty(join_type, batch_size, enable_aqe):
```

Comment:
nit: perhaps we can have a new test instead of reusing the existing tests? Specifically I don't see a lot of value in testing a join with a batch size of 100 and aqe on vs off.  The batch size is there to verify that the algorithm can handle lots of batches and that algorithm is not going to change internally based off of AQE.  Also the batch size of 100 is really slow to run.  If fact if you want a `test_right_broadcast_nested_loop_join_without_condition_empty_small_batch` test and remove the batch_size parameter from this one I would be fine with that. Because it would isolate the slowness to one specific test that is simpler to skip/separate.

This comment goes for all of the tests that you updated.